include/oneapi/tbb/detail/_task_handle.h

```diff
@@ -49,16 +78,59 @@ class task_handle_task : public d1::task {
     task_handle_task(d1::wait_tree_vertex_interface* vertex, d1::task_group_context& ctx, d1::small_object_allocator& alloc)
         : m_wait_tree_vertex(vertex)
         , m_ctx(ctx)
-        , m_allocator(alloc) {
+        , m_allocator(alloc)
+#if __TBB_PREVIEW_TASK_GROUP_EXTENSIONS
+        , m_dynamic_state(nullptr)
+#endif
+    {
         suppress_unused_warning(m_version_and_traits);
         m_wait_tree_vertex->reserve();
     }
 
     ~task_handle_task() override {
         m_wait_tree_vertex->release();
+#if __TBB_PREVIEW_TASK_GROUP_EXTENSIONS
+        task_dynamic_state* current_state = m_dynamic_state.load(std::memory_order_relaxed);
+        if (current_state != nullptr) {
+            current_state->release();
+        }
+#endif
     }
 
     d1::task_group_context& ctx() const { return m_ctx; }
+
+#if __TBB_PREVIEW_TASK_GROUP_EXTENSIONS
+    // Initializes the dynamic state if:
+    // * the task_completion_handle object was created
+    // * first dependency was added to a task_handle
+    // * Successors were transferred to the current task
+    task_dynamic_state* get_dynamic_state() {
```

Comment:
The comment seemingly describes the "expected" call sites for the function; that does not seem to add much value. A short description of what the function does would likely be better.

Reply:
Fixed